frontend/src/hooks/services.ts

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { CancelablePromise } from "../utils/CancelablePromises";
+import * as API from "../services/Api";
+import { DurationInSeconds, TimeInMilliseconds } from "../types/Common";
+import { ServiceDetailsInfo } from "../types/ServiceInfo";
+import { PeerAuthentication } from "../types/IstioObjects";
+import { AxiosError } from "axios";
+import { DecoratedGraphNodeData, NodeType } from "../types/Graph";
+import * as AlertUtils from "../utils/AlertUtils";
+import { useState } from "react";
+
+export function useServiceDetail(namespace: string, serviceName: string, duration?: DurationInSeconds, updateTime?: TimeInMilliseconds) {
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [fetchError, setFetchError] = React.useState<AxiosError | null>(null);
+
+  const [serviceDetails, setServiceDetails] = React.useState<ServiceDetailsInfo | null>(null);
+  const [gateways, setGateways] = React.useState<string[] | null>(null);
+  const [peerAuthentications, setPeerAuthentications] = React.useState<PeerAuthentication[] | null>(null);
+
+  React.useEffect(() => {
+    if (namespace.length === 0 || serviceName.length === 0) {
+      return;
+    }
+
+    setIsLoading(true); // Mark as loading
+    let getDetailPromise = API.getServiceDetail(namespace, serviceName, false, duration);
+    let getGwPromise = API.getIstioConfig('', ['gateways'], false, '', '');
```

Comment:
There was a change in loading of configs, it fails here:
![Screenshot from 2022-08-09 12-30-05](https://user-images.githubusercontent.com/604313/183629558-b06ca467-5bd8-443a-bfe1-7a1331847bf6.png)
Should call: "allIstioConfigs"

Reply:
@israel-hdez it seems another rebase is needed and fix of loading gateways, as Service dropdown is not loading.

Reply:
I just pushed the fix.
Hopefully, I did it right this time.